src/app/pages/masterdata/department.functionality/department.functionality.component.ts

```diff
@@ -55,7 +56,8 @@ export class DepartmentFunctionalityComponent implements OnInit {
       save(): void {
         let model = this.pagesForDepartment.filter(this.canViewd);
         let selectedDepartment = this.selectedDepartment;
-        this.pagePermissionModelToSave = model.map(function (this, data) {
```

Comment:
if you need some map function put it into service. Components should not have business logic